x-pack/legacy/plugins/ml/public/application/jobs/jobs_list/components/start_datafeed_modal/time_range_selector/time_range_selector.js

```diff
@@ -216,6 +217,57 @@ function TabStack({ title, items, switchState, switchFunc }) {
     </div>
   );
 }
+
+const DatePickerWithInput = ({ date, onChange, minDate, setIsValid, tab }) => {
+  const [dateString, setDateString] = useState(date.format(TIME_FORMAT));
+  const [currentTab, setCurrentTab] = useState(tab);
+
+  if (currentTab !== tab) {
+    // if the tab has changed, reset the text to be the same as the date prop
+    setDateString(date.format(TIME_FORMAT));
+    setCurrentTab(tab);
+  }
```

Comment:
I suspect this needs to go into a `useEffect()`. Even if it's guarded by the `if`, calling state setters shouldn't be done directly in the render method outside lifecycle hooks

Reply:
fixed in 55b913f2be1e22daeaa12e54df28e6753a9a1e50